pkg/image/registry/imagestreammapping/rest.go

```diff
@@ -64,13 +64,25 @@ func (s *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 		tag = api.DefaultImageTag
 	}
 
-	if err := s.imageRegistry.CreateImage(ctx, &image); err != nil && !errors.IsAlreadyExists(err) {
+	imageCreateErr := s.imageRegistry.CreateImage(ctx, &image)
+	if imageCreateErr != nil && !errors.IsAlreadyExists(imageCreateErr) {
 		return nil, err
 	}
 
+	// prefer dockerImageReference set on image for the tagEvent if the image is new
+	ref := image.DockerImageReference
+	if errors.IsAlreadyExists(imageCreateErr) && image.Annotations[api.ManagedByOpenShiftAnnotation] == "true" {
```

Comment:
This annotation is always checked against `"true"`, `kapi.ConditionTrue` is capitalized.